### PR TITLE
EZP-32110: Exposed more search result data from Pagefanta adapters

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/EZP-32110
| Versions      | master, 3.3

### CODE REVIEW

File: docs/api/public_php_api_search.md [hunk 1]
```diff
@@ -337,6 +340,7 @@ For more information and examples, see [PagerFanta documentation.](https://githu
 |[`LocationSearchHitAdapter`](https://github.com/ezsystems/ezplatform-kernel/blob/v1.0.0/eZ/Publish/Core/Pagination/Pagerfanta/LocationSearchHitAdapter.php)|Makes a Location search against passed Query and  returns [SearchHit](https://github.com/ezsystems/ezplatform-kernel/blob/v1.0.0/eZ/Publish/API/Repository/Values/Content/Search/SearchHit.php) objects instead.|
 |[`ContentFilteringAdapter`](https://github.com/ezsystems/ezplatform-kernel/blob/master/eZ/Publish/Core/Pagination/Pagerfanta/ContentFilteringAdapter.php)|Applies a Content filter and returns a `ContentList` object.|
 |[`LocationFilteringAdapter`](https://github.com/ezsystems/ezplatform-kernel/blob/master/eZ/Publish/Core/Pagination/Pagerfanta/LocationFilteringAdapter.php)|Applies a Location filter and returns a `LocationList` object.|
+|[`SearchResultAdapter`](https://github.com/ezsystems/ezplatform-kernel/blob/master/eZ/Publish/Core/Pagination/Pagerfanta/SearchResultAdapter.php)|Exposes additional search result data: aggregation results, max. score, computation time, and timeout flag.|
```
Comment (adamwojs) [inline docs/api/public_php_api_search.md:343]: This is just an interface implemented by `ContentSearchAdapter`, `ContentSearchHitAdapter`, `` `LocationSearchAdapter` ``, `LocationSearchHitAdapter` adapters. 

In practice this mean that only those could be used with `\eZ\Publish\Core\Pagination\Pagerfanta\Pagerfanta` and only those expose addition data. Ref. https://github.com/ezsystems/ezplatform-kernel/blob/243cdae6a8b788541c7637fe0b44b82b1ea0aea9/eZ/Publish/Core/Pagination/Pagerfanta/Pagerfanta.php#L16-L19

**Review by adamwojs — COMMENTED** [review]
Few comments:

* Could you please expand this section or add new one e.g. "Accessing additional search result data from Pagerfanta"  with more details and examples? 
* Link to `https://github.com/ezsystems/ezplatform-kernel/tree/v1.0.0/eZ/Publish/Core/Pagination/Pagerfanta` needs to be updated.

File: docs/api/public_php_api_search.md (hunk 1 above)
Comment (DominikaK) [reply docs/api/public_php_api_search.md:343]: So do I understand correctly that it makes to sense to list it in the table?

Comment (adamwojs) [reply docs/api/public_php_api_search.md:367]: Yes

**Review by dabrt — APPROVED** [review]

File: docs/api/public_php_api_search.md [hunk 2]
```diff
@@ -327,6 +327,33 @@ Pagination can then be rendered for example using the following template:
...
+- Computation time
+- Timeout flag
+
+``` php
+use eZ\Publish\Core\Pagination\Pagerfanta\Pagerfanta;
+
+class CustomController extends Controller
+{
+    //...
+    public function showContentAction(Request $request, $locationId)
```
Comment (adamwojs) [inline docs/api/public_php_api_search.md:345]:
```suggestion
    public function showContentAction(Request $request, int $locationId): Response
```

File: docs/api/public_php_api_search.md [hunk 3]
```diff
@@ -327,6 +327,33 @@ Pagination can then be rendered for example using the following template:
...
+
+You can access the following additional search result data from PagerFanta:
+
+- Aggregation results
+- Max. score
+- Computation time
+- Timeout flag
+
+``` php
+use eZ\Publish\Core\Pagination\Pagerfanta\Pagerfanta;
```
Comment (adamwojs) [inline docs/api/public_php_api_search.md:340]:
```suggestion
use eZ\Publish\Core\Pagination\Pagerfanta\Pagerfanta;
//...
```

**Review by adamwojs — APPROVED** [review]

**Pull request merged**